test/integration/h2_fuzz.cc

```diff
@@ -0,0 +1,215 @@
+#include "test/integration/h2_fuzz.h"
+
+#include <functional>
+
+#include "common/common/assert.h"
+#include "common/common/base64.h"
+#include "common/common/logger.h"
+
+#include "test/test_common/environment.h"
+
+namespace Envoy {
+
+using namespace Envoy::Http::Http2;
+
+void H2FuzzIntegrationTest::sendFrame(const test::integration::H2TestFrame& proto_frame,
+                                      std::function<void(const Http2Frame&)> write_func) {
+  Http2Frame h2_frame;
+  switch (proto_frame.frame_type_case()) {
+  case test::integration::H2TestFrame::kPing:
+    ENVOY_LOG_MISC(trace, "Sending ping frame");
+    h2_frame = Http2Frame::makePingFrame(proto_frame.ping().data());
+    break;
+  case test::integration::H2TestFrame::kSettings: {
+    const Http2Frame::SettingsFlags settings_flags =
+        static_cast<Http2Frame::SettingsFlags>(proto_frame.settings().flags());
+    ENVOY_LOG_MISC(trace, "Sending settings frame");
+    h2_frame = Http2Frame::makeEmptySettingsFrame(settings_flags);
+  } break;
+  case test::integration::H2TestFrame::kHeaders: {
+    const Http2Frame::HeadersFlags headers_flags =
+        static_cast<Http2Frame::HeadersFlags>(proto_frame.headers().flags());
+    uint32_t stream_idx = proto_frame.headers().stream_index();
+    ENVOY_LOG_MISC(trace, "Sending headers frame");
+    h2_frame = Http2Frame::makeEmptyHeadersFrame(stream_idx, headers_flags);
+  } break;
```

Comment:
Nit: prefer `break;` inside braces for readability.

Reply:
Done.